### PR TITLE
Alex latin1

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/tests/ParserTests/regressions/nothing-unicode.golden [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: 無
+version: 0
+cabal-version: >=1.10
+build-type: Simple
+license: UnspecifiedLicense
+synopsis: The canonical non-package 無
+x-無: 無
```
Comment (phadej) [inline Cabal/tests/ParserTests/regressions/nothing-unicode.golden]: ping @hvr 

It seems that ReadP parser accepts this. I'm 99% sure that all field-names on Hackage are ASCII  though. Should I add a `cabal chech`. Also what would be proper regexp for fieldnames. Currently it's *almost everything*:

```
$namecore        = [$printable] # [$space \: \" \{ \} $paren $symbol']
```

as field names are taken as is, but they can be validated further in the parser stage.

Comment (23Skidoo) [inline Cabal/tests/ParserTests/regressions/nothing-unicode.golden]: Surely we don't need to support Unicode field names?

Comment (phadej) [inline Cabal/tests/ParserTests/regressions/nothing-unicode.golden]: custom fields can be, but I added a check that we don't want them on  Hackage.

The story is very similar as with flag: French or German shop might have some `x-Straße` there. Let them have it, if they want.

Comment (23Skidoo) [inline Cabal/tests/ParserTests/regressions/nothing-unicode.golden]: I doubt that anyone uses this feature, but I can see no harm in supporting it.

**Pull request merged**